test/distributed/checkpoint/test_tp_checkpoint.py

```diff
@@ -0,0 +1,84 @@
+# Owner(s): ["oncall: distributed"]
+
+from copy import deepcopy
+
+import torch
+import torch.distributed.checkpoint as dist_cp
+import torch.nn as nn
+
+from torch.distributed._tensor import init_device_mesh
+
+from torch.distributed.checkpoint.default_planner import (
+    DefaultLoadPlanner,
+    DefaultSavePlanner,
+)
+
+from torch.distributed.tensor.parallel import PairwiseParallel, parallelize_module
+from torch.testing._internal.common_utils import run_tests
+
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    skip_if_lt_x_gpu,
+    with_comms,
+)
+from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+
+
+class ToyModel(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.net1 = nn.Linear(10, 32)
+        self.relu = nn.ReLU()
+        self.net2 = nn.Linear(32, 5)
+
+    def forward(self, x):
+        return self.net2(self.relu(self.net1(x)))
+
+
+class TestTpCheckpoint(DTensorTestBase):
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    @with_temp_dir
+    def test_tp_checkpoint(self):
+        CHECKPOINT_DIR = self.temp_dir
+        mesh_shpe = (self.world_size,)
+        tp_mesh = init_device_mesh(self.device_type, mesh_shpe)
+
+        # create model and move it to GPU with id rank
+        model = ToyModel().cuda(self.rank)
```

Comment:
Can we add a test case where we use Meta device?

Reply:
Will add that in follow up PR.